modules/nw-cluster-mtu-change.adoc

```diff
@@ -0,0 +1,482 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring-cluster-network-mtu.adoc
+
+[id="nw-cluster-mtu-change_{context}"]
+= Changing the cluster MTU
+
+As a cluster administrator, you can change the maximum transmission unit (MTU) for your cluster. The migration is disruptive and nodes in your cluster might be temporarily unavailable as the MTU update rolls out.
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+* You identified the target MTU for your cluster. The correct MTU varies depending on the cluster network provider that your cluster uses:
+** *OVN-Kubernetes*: The cluster MTU must be set to `100` less than the lowest MTU value in your cluster.
+** *OpenShift SDN*: The cluster MTU must be set to `50` less than the lowest MTU value in your cluster.
+
+.Procedure
+
+To increase or decrease the MTU for the cluster network complete the following procedure.
+
+. To obtain the current MTU for the cluster network, enter the following command:
++
+[source,terminal]
+----
+$ oc describe network.config cluster
+----
++
+.Example output
+[source,text]
+----
+...
+Status:
+  Cluster Network:
+    Cidr:               10.217.0.0/22
+    Host Prefix:        23
+  Cluster Network MTU:  1400
+  Network Type:         OpenShiftSDN
+  Service Network:
+    10.217.4.0/23
+...
+----
+
+. Prepare your configuration for the hardware MTU:
+
+** If your hardware MTU is specified in a NetworkManager connection configuration, complete the following steps. This approach is the default for {product-title} if you do not explicitly specify your network configuration with DHCP, a kernel command line, or some other method. Your cluster nodes must all use the same underlying network configuration for the following procedure to work unmodified.
+
+... To find the primary network interface, enter the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host ip route list match 0.0.0.0/0 | awk '{print $5 }'
+----
++
+--
+where:
+
+`<node_name>`:: Specifies the name of a node in your cluster.
+--
+
+... To find the connection profile that NetworkManager created for the interface name returned from the previous command, enter the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host nmcli c | grep <interface>
```

Comment:
For OVN-K, once you obtain the appropriate interface name as per my last comment above, this output will either:

* give three profiles, one is the one the node originally had if it had one configured, and the other two are the ones that we internally generate from the former, named `ovs-if-phys0` and `ovs-port-phys0`. The appropriate to use would be the the one the node originally had.
* give only two profiles that we internally generate named `ovs-if-phys0` and `ovs-port-phys0`. In this case, the node was never originally configured with a profile and one ephemeral default DHCP profile was once generated that might not longer be there. In this case, the customer has to use a default DHCP profile as a starting point.

Example output of the last case
```
❯ oc debug node/jcaamano-mtu-migratio-zxct2-master-0 -- chroot /host nmcli c | grep ens4                                   
Starting pod/jcaamano-mtu-migratio-zxct2-master-0-debug ...
To use host binaries, run `chroot /host`
ovs-if-phys0        353774d3-0d3d-4ada-b14e-cd4d8824e2a8  ethernet       ens4   
ovs-port-phys0      332ef950-b2e5-4991-a0dc-3158977c35ca  ovs-port       ens4  
```

Given the complexity, would we probably better suggest as an example in general to build a default DHCP profile with the MTU setting for a default (IPI) case, and mention for a non-default case (UPI) that the customer should customize the original deployed profile, which I guess should be aware of?

Reply:
Sorry for not noticing this before and all the extra work I might have caused.

Reply:
@jcaamano, that sounds unfortunately complicated. I'm not sure who might know how to accomplish this via DHCP, as I don't have any experience in the installation area. Maybe I can find someone, but probably not before docs freeze given the complexity in all of this.

Reply:
Looks like we have this for user-provisioned: https://docs.openshift.com/container-platform/4.9/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced_network_installing-bare-metal

Reply:
A default DHCP profile might look like this

```
[connection]
id=<id>
type=ethernet
autoconnect-priority=<0-99>
interface-name=<iface>

[ethernet]
mtu=<mtu>

[ipv4]
method=auto

[ipv6]
addr-gen-mode=stable-privacy
method=auto
```

But we can always refer to the appropriate RHEL NM documentation, for example:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configuring_ip_networking_with_nmcli#sec-Adding_and_Configuring_a_Dynamic_Ethernet_Connection_with_nmcli

Reply:
@jcaamano, I'll include this, thanks!